lib/repl.js

```diff
@@ -314,40 +308,42 @@ function REPLServer(prompt,
       debug('finish', e, ret);
       self.memory(cmd);
 
-      if (e && !self.bufferedCommand && cmd.trim().match(/^npm /)) {
-        self.outputStream.write('npm should be run outside of the ' +
-                                'node repl, in your normal shell.\n' +
-                                '(Press Control-D to exit.)\n');
-        self.bufferedCommand = '';
-        self.displayPrompt();
-        return;
-      }
-
       // If error was SyntaxError and not JSON.parse error
       if (e) {
-        if (e instanceof Recoverable) {
+        if (!self.bufferedCommand && cmd.trim().match(/^npm /)) {
+          self.outputStream.write('npm should be run outside of the ' +
+                                  'node repl, in your normal shell.\n' +
+                                  '(Press Control-D to exit.)\n');
+          self.bufferedCommand = '';
+          self.displayPrompt();
+        } else if (e instanceof Recoverable) {
           // Start buffering data like that:
           // {
           // ...  x: 1
           // ... }
           self.bufferedCommand += cmd + '\n';
           self.displayPrompt();
-          return;
         } else {
-          self._domain.emit('error', e);
+          debug('not recoverable, send to domain');
+          process.domain.emit('error', e);
```

Comment:
Instead of sometimes delegating to self._domain and sometimes to process.domain, always use process.domain. Note the evaluation function is bound to `self._domain` and the only way the `self._domain.emit` case could have been hit is via a non-recoverable error in `vm.createScript`, so there is no chance that a different domain could have been installed. (And even if there was, which domain to report the error to is underspecified so we should do the simplest thing possible)